vita3k/gui/src/imgui_impl_sdl_vulkan.cpp

```diff
@@ -0,0 +1,882 @@
+        -1, -1, 0, 1,
+    };
+
+    state.command_buffer.updateBuffer(state.transformation_buffer, 0, sizeof(matrix), matrix);
+
+    vk::RenderPassBeginInfo renderpass_begin_info(
+        state.renderpass, // Renderpass
+        state.framebuffers[image_index], // Framebuffer
+        vk::Rect2D(
+            vk::Offset2D(0, 0),
+            vk::Extent2D(state.get_renderer().swapchain_width, state.get_renderer().swapchain_height)),
+        1, &clear_value // Clear Colors
+    );
+
+    state.command_buffer.beginRenderPass(renderpass_begin_info, vk::SubpassContents::eInline);
+    state.command_buffer.bindPipeline(vk::PipelineBindPoint::eGraphics, state.pipeline);
+    state.command_buffer.bindDescriptorSets(
+        vk::PipelineBindPoint::eGraphics, // Bind Point
+        state.pipeline_layout, // Layout
+        0, // First Set
+        1, &state.matrix_set, // Sets
+        0, nullptr // Dynamic Offsets
+        );
+
+//    vk::Viewport viewport(0, 0, draw_data->DisplaySize.x, draw_data->DisplaySize.y, 0.0f, 1.0f);
+//    state.command_buffer.setViewport(0, 1, &viewport);
+
+    uint64_t vertex_offset_null = 0;
+
+    state.command_buffer.bindVertexBuffers(0, 1, &state.draw_buffer, &vertex_offset_null);
+    state.command_buffer.bindIndexBuffer(state.index_buffer, 0, imgui_index_type);
+
+    uint64_t vertex_offset = 0;
+    uint64_t index_offset = 0;
+
+    for (uint32_t a = 0; a < draw_data->CmdListsCount; a++) {
+        ImDrawList *draw_list = draw_data->CmdLists[a];
+
+        for (const auto &cmd : draw_list->CmdBuffer) {
+            if (cmd.UserCallback) {
+                cmd.UserCallback(draw_list, &cmd);
+            } else {
+                vk::Rect2D scissor_rect(
+                    vk::Offset2D(cmd.ClipRect.x, cmd.ClipRect.y),
+                    vk::Extent2D(cmd.ClipRect.z, cmd.ClipRect.w)
+                );
+                state.command_buffer.setScissor(0, 1, &scissor_rect);
+                auto *texture = reinterpret_cast<TextureState *>(cmd.TextureId);
+                state.command_buffer.bindDescriptorSets(
+                    vk::PipelineBindPoint::eGraphics, // Bind Point
+                    state.pipeline_layout, // Layout
+                    1,
+                    1, &texture->descriptor_set,
+                    0, nullptr
+                    );
+                state.command_buffer.drawIndexed(cmd.ElemCount, 1, index_offset, vertex_offset, 0);
+            }
+            index_offset += cmd.ElemCount;
+        }
+        
+        vertex_offset += draw_list->VtxBuffer.Size;
+    }
+
+    state.command_buffer.endRenderPass();
+
+    vk::ImageMemoryBarrier color_attachment_present_barrier(
+        vk::AccessFlagBits::eColorAttachmentWrite, // From rendering
+        vk::AccessFlagBits::eMemoryRead, // To readable format for presenting
+        vk::ImageLayout::eColorAttachmentOptimal,
+        vk::ImageLayout::ePresentSrcKHR,
+        VK_QUEUE_FAMILY_IGNORED, VK_QUEUE_FAMILY_IGNORED, // No Transfer
```

Comment:
Build is passing again :)

This is one thing we might want to get cleaned up. It seems to work without proper queue families on amd, nvidia and moltenvk. Is it okay to leave it like this?

Reply:
Sure, your choice. We can modify later if things turn bad.